src/main/java/net/dries007/tfc/objects/items/metal/ItemProspectorPick.java

```diff
@@ -74,7 +78,7 @@ public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos p
                 if (!targetStack.isEmpty())
                 {
                     // Just clicked on an ore block
-                    player.sendStatusMessage(new TextComponentTranslation("tfc.propick.found").appendText(" ").appendSibling(new TextComponentTranslation(targetStack.getTranslationKey() + ".name")), ConfigTFC.Client.TOOLTIP.propickOutputToActionBar);
+                    event = new ProspectEvent.Server(player, pos, ProspectEvent.ResultType.FOUND, targetStack.getTranslationKey(), 0);
```

Comment:
I like the work you put into this, it's a clean implementation, but...

You could have just done this on the world.isRemote branch, and avoid sending packets.

Reply:
No. Vein information shouldn't be required on both sides.

Reply:
Server side does game logic, client side does presentation. There's no argument there.